### PR TITLE
feat(compute): add compute reservation create shared

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:72]: In the test envaironment we don't have possibility to create additional projects to implement this feature. For this reason we have different implementation for this sample. And for test we mocked everything.

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:57]: Thank you!

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java
Comment (m-strzelczyk) [reply compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:72]: Please indicate, those CONSUMER_PROJECT_ID_1 and _2 are the projects that will have access to the reservation.

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java
Comment (m-strzelczyk) [inline compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:45]: Make sure to highlight the fact, that this project MUST be allowlisted in `compute.sharedReservationsOwnerProjects` org policy. See more at: https://cloud.google.com/compute/docs/instances/reservations-shared#shared_reservation_constraint

File: compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java
Comment (m-strzelczyk) [inline compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java:260]: Add explenation why we need to mock + what would have to be changes if someone wants to test this code without mocking.

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]

**Review by Sita04 — APPROVED** [review]
One nit. LGTM.
Kindly get Maciej's LGTM before merging.

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**
